### DISCUSSION TITLE
Nullability convention for Main()

### DISCUSSION THREAD

**Discussion by iSazonov** [dotnet/csharplang#3780] [Q&A] [answered]
I could not find a documented convention what is right nullable annotation for C# Main() entry point:
```c#
        public static int Main(string[] args);
        public static int Main(string?[] args);
        public static int Main(string[]? args);
        public static int Main(string?[]? args);
```

(Side question: in 3 and 4 lines syntax highlighting does not work in GitHub and VS Code - where to report? Roslyn repo?

[a comment by an automated account is not shown]

**Comment by jkotas** [comment, marked as answer]
It should be the first one: `public static int Main(string[] args);`

**Comment by iSazonov** [comment]
Thanks! Could you please convert this in a _documentation_ issue?

**Comment by canton7** [comment]
It might be better, I think, to find the documentation page you think should be updated, and then use the "Feedback" section at the bottom. That submits an issue to the right repo, and also adds in a bunch of tracking information to link the issue to the right doc page.

**Comment by PathogenDavid** [comment]
> Side question: in 3 and 4 lines syntax highlighting does not work in GitHub and VS Code

GitHub uses the same grammar set as [Linguist](https://github.com/github/linguist) for markdown code blocks which specifies [atom/language-csharp](https://github.com/atom/language-csharp) for C# syntax highlighting which in turn uses [dotnet/csharp-tmLanguage](https://github.com/dotnet/csharp-tmLanguage) which in turn has this issue: https://github.com/dotnet/csharp-tmLanguage/issues/157 which has a linked PR https://github.com/dotnet/csharp-tmLanguage/pull/175 that has gone stale for reasons that are not apparent.

I imagine csharp-tmLanguage is what vscode uses as well, but according to https://github.com/OmniSharp/omnisharp-vscode/issues/3404 you can enable the [Semantic Highlighting Preview](https://github.com/OmniSharp/omnisharp-vscode#semantic-highlighting) to get better highlighting.

TL;DR: Turn on semantic highlighting in vscode and subscribe to https://github.com/dotnet/csharp-tmLanguage/issues/157

**Comment by Youssef1313** [comment]
Depends on whether `Main` is called by any user-code or not.
